tiled/client/stream.py

```diff
@@ -259,6 +266,8 @@ def __init__(
         self._disconnect_lock = threading.Lock()
         self._disconnect_event = threading.Event()
         self._thread = None
+        self._last_received_sequence = None  # Track last sequence for reconnection
+        self._connected = False  # Track connection state
```

Comment:
Under the general rule of "state is the root of all evil" I wonder if we can avoid adding this state. I think that the control flow ensures that `self._connect(...)` only gets called (1) when `self._run(...)` starts and (2) after a connection error is raised. So the `if self._connected: return` cut-out is protecting a codepath that, in fact, you can never hit.